E2E/e2eTests/Source/Abilities/Sdk.swift

```diff
@@ -0,0 +1,221 @@
+                    storeType: .memory
+                ),
+                keychain: KeychainMock()
+            )).build()
+            let pollux = PolluxBuilder().build()
+            let mercury = MercuryBuilder(
+                castor: castor,
+                secretsStream: Client.createSecretsStream(
+                    keyRestoration: apollo,
+                    pluto: pluto,
+                    castor: castor
+                )
+            ).build()
+            
+            PrismAgent.setupLogging(logLevels: [
+                .prismAgent: .warning
+            ])
+            
+            prismAgent = PrismAgent(
+                apollo: apollo,
+                castor: castor,
+                pluto: pluto,
+                pollux: pollux,
+                mercury: mercury,
+                mediationHandler: BasicMediatorHandler(
+                    mediatorDID: mediatorDID,
+                    mercury: mercury,
+                    store: BasicMediatorHandler.PlutoMediatorStoreImpl(pluto: pluto)
+                )
+            )
+        }
+        
+        func initialize(_ actor: Actor) async throws {
+            try await prismAgent.start()
+            
+            prismAgent.handleReceivedMessagesEvents()
+                .sink(
+                    receiveCompletion: { completion in
+                        switch completion {
+                        case .finished:
+                            print("Publisher completed successfully.")
+                        case .failure(let error):
+                            print("Publisher failed with error: \(error)")
+                        }
+                    },
+                    receiveValue: { message in
+                        if (self.receivedMessages.contains(message.id)) {
+                            return
+                        }
+                        self.receivedMessages.append(message.id)
+                        switch(message.piuri) {
+                        case ProtocolTypes.didcommOfferCredential3_0.rawValue:
+                            self.credentialOfferStack.append(message)
+                        case ProtocolTypes.didcommIssueCredential3_0.rawValue:
+                            self.issueCredentialStack.append(message)
+                        case ProtocolTypes.didcommRequestPresentation.rawValue:
+                            self.proofOfRequestStack.append(message)
+                        default:
+                            break
+                        }
+                    }
+                )
+                .store(in: &cancellables)
+            
+            prismAgent.startFetchingMessages()
+        }
+        
+        func tearDown() async throws {
+            prismAgent.stopFetchingMessages()
+            try await prismAgent.stop()
+        }
+        
+        static private func getPrismMediatorDid() async throws -> DID {
+            let url = URL(string: Config.mediatorOobUrl)!
```

Comment:
Usually this force unwraps (`!`)  are bad but since it is a testing framework it doesn't matter :)

Reply:
Yeah, we could throw an exception if the property is not found but this way was easier 😅